uxarray/core/dataarray.py

```diff
@@ -358,3 +360,62 @@ def _edge_centered(self) -> bool:
         """Returns whether the data stored is Edge Centered (i.e. contains the
         "n_edge" dimension)"""
         return "n_edge" in self.dims
+
+    def isel(self, ignore_grid=False, *args, **kwargs):
+        """Grid-informed implementation of xarray's ``isel`` method, which
+        enables indexing across grid dimensions (a.k.a.
+
+        ``n_node``, ``n_edge``, or ``n_face``)
+        """
+
+        from uxarray.constants import GRID_DIMS
+
+        if any(grid_dim in kwargs
+               for grid_dim in GRID_DIMS) and not ignore_grid:
+            # slicing a grid-dimension through grid object
+
+            dim_mask = [grid_dim in kwargs for grid_dim in GRID_DIMS]
+            dim_count = np.count_nonzero(dim_mask)
+
+            if dim_count > 1:
+                raise ValueError(
+                    "Only one grid dimension can be sliced at a time")
+
+            if "n_node" in kwargs:
+                sliced_grid = self.uxgrid.isel(n_node=kwargs['n_node'])
+            elif "n_edge" in kwargs:
+                sliced_grid = self.uxgrid.isel(n_edge=kwargs['n_edge'])
+            else:
+                sliced_grid = self.uxgrid.isel(n_face=kwargs['n_face'])
+
+            return self._slice_from_grid(sliced_grid)
+
+        else:
+            # original xarray implementation for non-grid dimensions
+            return super().isel(*args, **kwargs)
+
+    def _slice_from_grid(self, sliced_grid):
```

Comment:
@erogluorhan 

Regarding your comment, this is the internal function used for generating the new DataArray.

Reply:
Got it!